assets/js/state/sagas/settings.test.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { recordSaga } from '@lib/test-utils';
+
+import { networkClient } from '@lib/network';
+import MockAdapter from 'axios-mock-adapter';
+
+import { apiKeySettingsFactory } from '@lib/test-utils/factories/settings';
+import { addDays, formatISO, subDays } from 'date-fns';
+import HealthIcon from '@common/HealthIcon';
+import { notify, dismissableNotify } from '@state/notifications';
+import { checkApiKeyExpiration } from './settings';
+
+const axiosMock = new MockAdapter(networkClient);
+
+describe('Settings sagas', () => {
+  beforeEach(() => {
+    axiosMock.reset();
+  });
+
+  describe('checkApiKeyExpiration saga', () => {
+    it('should skip any expiration toast creation when the expire_at of the api key is not defined', async () => {
+      axiosMock
+        .onGet('/api/v1/settings/api_key')
+        .reply(200, apiKeySettingsFactory.build({ expire_at: null }));
+
+      const dispatched = await recordSaga(checkApiKeyExpiration, {});
+
+      expect(dispatched).toEqual([]);
+    });
+
+    it('should dispatch an expired key notification if the expire_at of the api key is in the past', async () => {
+      axiosMock.onGet('/api/v1/settings/api_key').reply(
+        200,
+        apiKeySettingsFactory.build({
+          expire_at: formatISO(subDays(new Date(), 2)),
+        })
+      );
+
+      const dispatched = await recordSaga(checkApiKeyExpiration, {});
+      const expectedAction = notify({
+        text: 'API Key has expired. Go to Settings to issue a new key',
+        icon: <HealthIcon health="critical" />,
+        duration: Infinity,
+        id: 'api-key-expiration-toast',
+      });
+      expect(dispatched).toEqual([expectedAction]);
+    });
+
+    it('should dispatch an expires in dismissable notification if the api key expiration days are less then 30', async () => {
+      axiosMock.onGet('/api/v1/settings/api_key').reply(
+        200,
+        apiKeySettingsFactory.build({
+          expire_at: formatISO(addDays(new Date(), 3)),
+        })
+      );
+
+      const dispatched = await recordSaga(checkApiKeyExpiration, {});
+      const expectedAction = dismissableNotify({
+        text: `API Key expires in 2 days`,
+        icon: <HealthIcon health="warning" />,
+        duration: Infinity,
+        id: 'api-key-expiration-toast',
+      });
+      expect(dispatched).toEqual([expectedAction]);
+    });
```

Comment:
New line here, after the test is finished

Reply:
done